### PR TITLE
<feat>(rc.cc): new get_config option (#290)

### PR DESCRIPTION
**Pull request by Shea690901**
make ALL configured ports visible to mudlib

### CODE REVIEW

**Comment by Shea690901** [comment]
Ups…
Typo…
One shouldn't necessarily code when one is too tired, but this version should work ;)

**Comment by Shea690901** [comment]
read issue #290 before merge!

File: src/include/runtime_config.h [hunk 1]
```diff
@@ -35,11 +35,12 @@
 #define __DEFAULT_FAIL_MESSAGE__ CFG_STR(12)
 #define __GLOBAL_INCLUDE_FILE__ CFG_STR(13)
 #define __MUD_IP__ CFG_STR(14)
+#define __PORT_CONFIG__ CFG_STR(15)
 
 /*
  * These config settings return an int (ie number)
  */
-#define BASE_CONFIG_INT (BASE_CONFIG_STR + 15)
+#define BASE_CONFIG_INT (BASE_CONFIG_STR + 100)
```
Comment (thefallentree) [inline src/include/runtime_config.h]: This change will cause lib to copy this header again to get correct position.

Comment (Shea690901) [inline src/include/runtime_config.h]: Huh?
What do you mean?

**Comment by Shea690901** [comment]
Sadly my tablet broke...
Meaning: currently I not only can´t compile the driver myself, I also can´t code :(

On a nicer note:
If everything goes well, I´ll get a new laptop(!) within the next few weeks :)

**Comment by thefallentree** [comment]
RC.cc will probably be ported into GO code real soon, just so you know

**Comment by Shea690901** [comment]
completly unknown language :(

**Comment by cartercheng** [comment]
I wonder if this might be a concern for the future. Having to learn two languages(maybe 3 including lpc) to maintain the driver is obviously more work than needing to know just one(or two). Unless we port the entire driver into go or something like that.

**Comment by thefallentree** [comment]
well， you can always work on just part of the driver. We will break driver
apart, and make each part self contained

**Comment by Shea690901** [comment]
In this case the breaking should be the very first act...
Before any code is changed (apart from changes reflecting now changed pathes)!

File: src/include/runtime_config.h (hunk 1 above)
Comment (thefallentree) [inline src/include/runtime_config.h]: by changing the number here from 15 to 100 , you effectively changed the argument that people use set_config()  and get_config() .  Thus this is a compatibility issue.

I think a better way to implment this is to actually implement double ordering. separating user visible number with underlying storage index

Comment (thefallentree) [inline src/include/runtime_config.h]: @thefallentree
thefallentree 18 seconds ago Member
by changing the number here from 15 to 100 , you effectively changed the argument that people use set_config() and get_config() . Thus this is a compatibility issue.

I think a better way to implment this is to actually implement double ordering. separating user visible number with underlying storage index

**Comment by thefallentree** [comment]
@Shea690901  Ideally, I agree with you. But, there is a bit of chicken & egg issue,  I don't know exactly how to untangle the mess before I implement what I want.  So..it's a WIP , but you are right, we need to define better modules, packages in the codebase.

File: src/include/runtime_config.h (hunk 1 above)
Comment (Shea690901) [inline src/include/runtime_config.h]: I would say, when somebody uses `CFG_STR(20)` and expects to get `__COMPILER_STACK_SIZE__ `, then it´s his own problem...
Alas, this is the very reason for this include file...
I know:
- which values are available
- which value results in what return type
- what the meaning is behind which value
- and (most important) which name I can use to get the desired result, regardless of driver changes

**Comment by Shea690901** [comment]
Since this is (planned to) change to go I´m closing this pr...
Although I´ll keep the feature branch for now as reference.

**Pull request closed without merging**